feat(useMeasure): support custom measurers in order to support border-box and others

m-kolomoyets opened the pull request:
### What is the current behavior, and the steps to reproduce the issue?

Current implementation of the `useMeasure` hook calculates the element bounds based on content-box model only. 

It is a little unexpected behavior when the project and all the layout is based on border-box model and trying to calculate measures of the element has paddings. 

Thus, current hook ignores paddings calculating element's width and height.

### What is the expected behavior?

Giving the possibility to define custom measure matcher model to take the values needed.

### How does this PR fix the problem?

By adding the additional optional function argument receives the `ResizeObserverEntry` value and returns the new measures should be applied or `null`. If function returns `null`, the default behavior is applied and content-box measures will be returned.

## Checklist

- [x] Have you read [contribution guideline](../../CONTRIBUTING.md)?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Is there an existing issue for this PR?
  - https://github.com/react-hookz/web/issues/1548
- [x] Have the files been linted and formatted?
- [x] Have the docs been updated to match the changes in the PR?
- [x] Have the tests been updated to match the changes in the PR?
- [x] Have you run the tests locally to confirm they pass?

m-kolomoyets committed:
feat(use-measure): extended useMeasure hook to custom measures define matcher

m-kolomoyets committed:
docs(use-measure): added story and extended existing documentation with new argument

m-kolomoyets committed:
fix(yarn.lock): yarn.lock formatting return

m-kolomoyets committed:
fix(default-measure): added defaultMeasures variable to all scenarios

xobotyi commented on src/useMeasure/index.ts:
```diff
@@ -12,9 +12,11 @@ export type Measures = {
  */
 export function useMeasure<T extends Element>(
 	enabled = true,
+	observerEntryMatcher?: (entry: ResizeObserverEntry) => Measures | null,
```
Proper solution here would be:
1) to define the type `type Measurer = (entry: ResizeObserverEntry) => Measures`. note there is no option for measure to return `null` as you did, since in case element exists it is measurable (a don't see option for etry to me non-measurable)
2) expose two implementations of measurer - `contentBoxMeasurer` and `borderBoxMeasurer`
3) add parameter that defaults to `contentBoxMeasurer`
4) use that measurer all the time

xobotyi commented:
Thanks for contribution!

m-kolomoyets committed:
fix(review): added measurers functions to comfy enable different box sizing options

xobotyi reviewed (approved):
LGTM

m-kolomoyets committed:
feat(useMeasure): accept custom measurer

Measures came from the content box only, so consumers laying out in
border-box read sizes without paddings and borders. A second parameter
takes a measurer deriving `Measures` from the observer entry:
`contentBoxMeasurer` stays the default, `borderBoxMeasurer` ships next
to it for opt-in border-box sizing.

Re-measurement still rides on the shared observer's default box, so a
padding-only change that leaves the content box intact emits no new
measurement.

Closes: #1548
